src/libraries/System.Text.Json/src/System/Text/Json/TypeExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace System.Text.Json
+{
+    internal static class TypeExtensions
+    {
+        private static readonly Type s_nullableOfTType = typeof(Nullable<>);
+
+        /// <summary>
+        /// Returns <see langword="true" /> when the given type is of type <see cref="Nullable{T}"/>.
+        /// </summary>
+        public static bool IsNullableValueType(this Type type)
+        {
+            return type.IsGenericType && type.GetGenericTypeDefinition() == s_nullableOfTType;
```

Comment:
I believe this can just be:
```cs
return Nullable.GetUnderlyingType(type) != null;
```

This is what we do in https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/NullableConverterFactory.cs#L11-L14